src/appmixer/utils/controls/OnError/OnError.js

```diff
@@ -1,59 +1,94 @@
 'use strict';
 
-async function getErrors(context, now, errors = [], from = 0, size = 100) {
-
-    const result = await context.callAppmixer({
-        endPoint: '/logs',
-        method: 'GET',
-        qs: {
-            from,
-            size,
-            sort: 'gridTimestamp:asc',
-            query: `@timestamp:[${context.state.since} TO ${now}] AND severity:error`,
-            flowId: context.flowId
-        }
-    });
+async function fetchErrorsPage(context, lock, limit, logId = null, lastGridTimestamp = null, size = 100, errors = []) {
+
+    const qs = {
+        size: Math.min(limit, size),
+        query: 'severity:error',
+        flowId: context.flowId
+    };
+
+    if (logId && lastGridTimestamp) {
+        qs.searchAfter = [logId, lastGridTimestamp];
+    }
+
+    while (true) {
+        try {
+            const result = await context.callAppmixer({
+                endPoint: '/logs',
+                method: 'GET',
+                qs
+            });
+
+            if (!(result.hits && result.hits.length > 0 && limit > result.hits.length)) {
+                break;
+            }
 
-    errors = errors.concat(result.hits);
+            lock.extend(parseInt(context.config.lockTTL, 10) || 1000 * 60 * 2);
+            const lastLog = result.hits[result.hits.length - 1];
+            errors.push(...result.hits);
 
-    if (result.hits.length === 0 || result.hits.length < size) {
-        return errors;
+            if (errors.length >= limit) {
+                break;
+            }
+
+            qs.searchAfter = [lastLog['_id'], lastLog['gridTimestamp']];
+        } catch (error) {
+            console.error('Error fetching errors:', error);
```

Comment:
REmove console.error. That can just cause troubles. You can keep the context.log.
Axios, for example, has huge exceptions, if you put them to stderr/stdout then logstash splits them line by line and creates tons of logs in Elastic.